Tests/UpstreamIntegrationTests/UpstreamIntegrationTests.swift

```diff
@@ -69,52 +66,495 @@ class UpstreamIntegrationTests: TestBase {
+                {
+                  "maxAge": 123,
+                  "key": "kndctr_972C898555E9F7BC7F000101_AdobeOrg_cluster",
+                  "value": "\#(locationHint)"
+                }
+              ]
+            }
+            """#
+            
+            // See testSendEvent_receivesExpectedEventHandles for existence validation
+            let stateStoreEvent = getEdgeEventHandles(expectedHandleType: TestConstants.EventSource.STATE_STORE).last!
+            
+            // Exact match used here to strictly validate `payload` array element count == 2
+            assertExactMatch(expected: getAnyCodable(expectedStateStoreJSON)!,
+                            actual: getAnyCodable(stateStoreEvent),
+                            typeMatchPaths: ["payload[0].maxAge"])
+            
+            resetTestExpectations()
+            networkService.reset()
+        }
+    }
+    
+
+    // MARK: 2nd event tests
+    func testSendEventx2_receivesExpectedNetworkResponse() {
+        // Setup
+        // These expectations are used as a barrier for the event processing to complete
+        expectEdgeEventHandle(expectedHandleType: TestConstants.EventSource.LOCATION_HINT_RESULT)
+
+        let experienceEvent = ExperienceEvent(xdm: ["xdmtest": "data"],
+                                              data: ["data": ["test": "data"]])
+        
+        Edge.sendEvent(experienceEvent: experienceEvent)
+        
+        // Extract location hint from Edge Network location hint response event
+        guard let locationHintResult = getLastLocationHintResultValue() else {
+            XCTFail("Unable to extract valid location hint from location hint result event handle.")
+            return
+        }
+        
+        // Wait on all expectations to finish processing before clearing expectations
+        assertExpectedEvents(ignoreUnexpectedEvents: true)
+        
+        // Reset all test expectations
+        networkService.reset()
+        resetTestExpectations()
+        
+        // Set actual testing expectations
+        // If test suite level location hint is not set, uses the value extracted from location hint result
+        let locationHintNetworkRequest = NetworkRequest(urlString: createURLWith(locationHint: edgeLocationHint?.rawValue ?? locationHintResult), httpMethod: .post)!
+        networkService.setExpectationForNetworkRequest(networkRequest: locationHintNetworkRequest, expectedCount: 1)
+        
+        // Test
+        // 2nd event
+        Edge.sendEvent(experienceEvent: experienceEvent)
+        
+        // Verify
+        // Network response assertions
+        let matchingResponses = networkService.getResponsesFor(networkRequest: locationHintNetworkRequest, timeout: 5)
+        
+        XCTAssertEqual(1, matchingResponses.count)
+        XCTAssertEqual(200, matchingResponses.first?.responseCode)
+
+    }
+    
+    func testSendEventx2_receivesExpectedEventHandles() {
+        sendEventAndResetExpectations()
```

Comment:
What do you think, since we are only asserting on the handle types, maybe we don't need sendEventAndResetExpectations and test like:
```
expectEdgeEventHandle(expectedHandleType: TestConstants.EventSource.LOCATION_HINT_RESULT, expectedCount: 2)
expectEdgeEventHandle(expectedHandleType: TestConstants.EventSource.STATE_STORE, expectedCount: 2)

Edge.sendEvent(..)
Edge.sendEvent(..)

assertExpectedEvents()

Reply:
Looks great, can we follow the same approach in `testSendEventx2_doesNotReceivesErrorEvent` and remove `sendEventAndResetExpectations`